app/testOnly/models/formsmodel/StartJourneyForm.scala

```diff
@@ -67,25 +68,81 @@ object StartJourneyForm {
+    )
+
+    val debtTotalAmountFormat: Formatter[BigDecimal] =
+      new Formatter[BigDecimal] {
+        override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], BigDecimal] =
+          taxRegimeFormatter.bind(taxRegimeKey, data).flatMap{ taxRegime =>
+            val (maxAmount, amountKey) = taxRegime match {
+              case TaxRegime.Epaye => payeMaxAmountOfDebt -> payeDebtTotalAmountKey
+              case TaxRegime.Vat   => vatMaxAmountOfDebt -> vatDebtTotalAmountKey
+            }
+            val minAmount = AmountInPence(100)
+            val dataWithDefaultValue = if (data.get(amountKey).exists(_.nonEmpty)) data else (data.updated(amountKey, "1234.53"))
+
+            amountOfMoneyFormatter(
+              isTooSmall = minAmount > AmountInPence(_),
+              isTooLarge = AmountInPence(_) > maxAmount
+            ).bind(amountKey, dataWithDefaultValue).leftMap(errors =>
+              errors.map { e =>
+                val mappedMesage = e.message match {
+                  case "error.pattern"  => "Total debt amount must be a number"
+                  case "error.required" => "Total debt amount not found"
+                  case "error.tooLarge" => s"Total debt amount must be below ${maxAmount.gdsFormatInPounds}"
+                  case "error.tooSmall" => s"Total debt amount must be above ${minAmount.gdsFormatInPounds}"
+                  case other            => other
+                }
+                FormError(e.key, mappedMesage)
+              })
+          }
+
+        override def unbind(key: String, value: BigDecimal): Map[String, String] =
+          Map(key -> formatAmountOfMoneyWithoutPoundSign(value))
+      }
+
+    val taxReferenceFormat: Formatter[TaxId] =
+      new Formatter[TaxId] {
+        override def bind(key: String, data: Map[String, String]): Either[Seq[FormError], TaxId] =
+          taxRegimeFormatter.bind(taxRegimeKey, data).map{ taxRegime =>
+            val (taxReferenceKey, defaultTaxRef): (String, TaxId) = taxRegime match {
+              case TaxRegime.Epaye => payeTaxReferenceKey -> RandomDataGenerator.nextEpayeRefs()(Random)._3
+              case TaxRegime.Vat   => vatTaxReferenceKey -> RandomDataGenerator.nextVrn()(Random)
+            }
+            val dataWithDefaultValue: Map[String, String] =
+              if (data.get(taxReferenceKey).exists(_.nonEmpty)) data else data.updated(taxReferenceKey, defaultTaxRef.value)
+
+            val taxRef: TaxId = taxRegime match {
+              case TaxRegime.Epaye => EmpRef(dataWithDefaultValue.getOrElse(taxReferenceKey, defaultTaxRef.value))
+              case TaxRegime.Vat   => Vrn(dataWithDefaultValue.getOrElse(taxReferenceKey, defaultTaxRef.value))
+            }
+            taxRef
+          }
+
+        // we don't actually need the unbind, but because we've created a Formatter, we need to declare one...
+        override def unbind(key: String, value: TaxId): Map[String, String] = ???
```

Comment:
the `unbind` is needed to transform from the case class to the underling data map in the form - this is typically used when prepopulating the data on the page. It should be easy to write this one though - you just look at the tax id type to determine if the key should be the PAYE or VAT one and create a map where that key maps onto the value of the tax id.

Reply:
ok